src/tests/EpicsConnectionStatusTests.cpp

```diff
@@ -0,0 +1,166 @@
+TEST_F(Schema_ep00, FileInitOk) {
+  ep00::HDFWriterModule Writer;
+  EXPECT_TRUE(Writer.init_hdf(UsedGroup, "{}") ==
+              HDFWriterModule_detail::InitResult::OK);
+  ASSERT_TRUE(RootGroup.has_group(NXLogGroup));
+  auto TestGroup = RootGroup.get_group(NXLogGroup);
+  EXPECT_TRUE(TestGroup.has_dataset(StatusName));
+  EXPECT_TRUE(TestGroup.has_dataset(TimestampName));
+}
+
+TEST_F(Schema_ep00, ReopenFile) {
+  ep00::HDFWriterModule Writer;
+  EXPECT_FALSE(Writer.reopen(UsedGroup) ==
+               HDFWriterModule_detail::InitResult::OK);
+}
+
+TEST_F(Schema_ep00, FileInitFail) {
+  ep00::HDFWriterModule Writer;
+  EXPECT_TRUE(Writer.init_hdf(UsedGroup, "{}") ==
+              HDFWriterModule_detail::InitResult::OK);
+  EXPECT_FALSE(Writer.init_hdf(UsedGroup, "{}") ==
+               HDFWriterModule_detail::InitResult::OK);
+}
+
+TEST_F(Schema_ep00, ReopenFileSuccess) {
+  ep00::HDFWriterModule Writer;
+  EXPECT_TRUE(Writer.init_hdf(UsedGroup, "{}") ==
+              HDFWriterModule_detail::InitResult::OK);
+  EXPECT_TRUE(Writer.reopen(UsedGroup) ==
+              HDFWriterModule_detail::InitResult::OK);
+}
+
+TEST_F(Schema_ep00, WriteDataSuccess) {
+
+  size_t BufferSize;
+  uint64_t Timestamp = 5555555;
+  std::string SourceName = "SIMPLE:DOUBLE";
+  auto Status = EventType::CONNECTED;
+  std::unique_ptr<std::int8_t[]> Buffer =
+      GenerateFlatbufferData(BufferSize, Timestamp, Status, SourceName);
+  ep00::HDFWriterModule Writer;
+  {
+    EXPECT_TRUE(Writer.init_hdf(UsedGroup, "{}") ==
+                HDFWriterModule_detail::InitResult::OK);
+    EXPECT_TRUE(Writer.reopen(UsedGroup) ==
+                HDFWriterModule_detail::InitResult::OK);
+  }
+  FileWriter::FlatbufferMessage TestMsg(
+      reinterpret_cast<const char *>(Buffer.get()), BufferSize);
+  EXPECT_NO_THROW(Writer.write(TestMsg));
+  auto TimeDataSet = UsedGroup.get_dataset(TimestampName);
+  auto Size = TimeDataSet.dataspace().size();
+  std::vector<uint64_t> Timestamps(Size);
+  TimeDataSet.read(Timestamps);
+  EXPECT_EQ(Timestamps[0], Timestamp);
+
+  auto StatusDataset = UsedGroup.get_dataset(StatusName);
+  std::vector<std::string> StatusData(StatusDataset.dataspace().size());
+  auto Datatype = hdf5::datatype::String::fixed(20);
+  Datatype.encoding(hdf5::datatype::CharacterEncoding::UTF8);
+  auto Dataspace = hdf5::dataspace::Simple({1});
+  EXPECT_NO_THROW(StatusDataset.read(StatusData, Datatype, Dataspace));
+  std::string StringFromDataset = StatusData[0];
+  removeTrailingNullFromString(StringFromDataset);
+  EXPECT_EQ(StringFromDataset, "CONNECTED");
+}
+
+TEST_F(Schema_ep00, FBReaderNoSourceName) {
+  size_t BufferSize;
+  uint64_t Timestamp = 5555555;
+  std::string SourceName;
+  auto Status = EventType::NEVER_CONNECTED;
+  std::unique_ptr<std::int8_t[]> Buffer =
+      GenerateFlatbufferData(BufferSize, Timestamp, Status, SourceName);
+  ep00::HDFWriterModule Writer;
+  EXPECT_TRUE(Writer.init_hdf(UsedGroup, "{}") ==
```

Comment:
This is repeating the expectations from FileInitOk and ReopenFileSuccess.
That is not usually a good thing.